File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public int[] moveZeroes(int[] nums) {
        int l = 0;
        int r = 0;
        int n = nums.length;

        while (r < n) {
            if (nums[l] == 0 && nums[r] != 0) {
                int tmp = nums[r];
                nums[r] = nums[l];
                nums[l] = tmp;
                l++;
            }
            if (nums[l] != 0) {
                l++;
            }
            r++;
        }

        return nums;
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
         l = 0
         r = 0
         n = len(nums)

         while r < n: 
            if nums[l] == 0 and nums[r] != 0: 
                
                nums[l] = nums[r]
                nums[r] = 0
                l +=1
            
            if  nums[l] != 0:
                l +=1
            
            r +=1
                
            
        
        